Usb.cpp

```diff
@@ -823,7 +823,12 @@ uint8_t USB::getConfDescr(uint8_t addr, uint8_t ep, uint8_t conf, USBReadParser
 
         //USBTRACE2("\r\ntotal conf.size:", total);
 
-        return ( ctrlReq(addr, ep, bmREQ_GET_DESCR, USB_REQUEST_GET_DESCRIPTOR, conf, USB_DESCRIPTOR_CONFIGURATION, 0x0000, total, bufSize, buf, p));
+        /*
+	   At least 045e:0289 complains if nbytes is greater than total when calling ctrlReq().
+           Make sure that we don't request chunks greater than total length, now that XBOXOLD
+	   retrieves and parses configuration descriptors.
+	 */
```

Comment:
Won't it be better to increase `bufSize`?

This code would get an incomplete configuration descriptor if `total` is greater than `bufSize`.

Reply:
Maybe my comment is not clear, I'll try to explain what's the issue.

The problem is that the device errors out if you try to read more bytes than the total configuration descriptor total length.

The normal way of reading such a descriptor is you first read the configuration descriptor with a small enough length (minimum 9 bytes is required because some controllers complain if you read less than the basic configuration descriptor information size) to retrieve the actual total length `wTotalLength` of the configuration descriptor. Then, once you know the total length, you read again the full configuration descriptor using the total length.

But, and here's the thing, if you request more data than the total configuration descriptor length, the device complains (and the current code tries to read 64 bytes when the configuration descriptor is just 12 bytes).

So the fix is to not read more than the total configuration descriptor length (and of course we need a buffer properly sized).

The readBuf size is 64 bytes, the total configuration descriptor length is 12 bytes in this case, so we need to use the readBuf but limit the total data requested to 12 bytes for the control request to work.

Increasing the buffer size would help only in case we know that we need to read a larger than 64 bytes configuration descriptor.
With the current proposed fix it is true that we could get an incomplete configuration descriptor if `total` is greater than `bufSize`, as we won't be able to read the full descriptor in the buffer. But that is a consequence of the finite buffe size (and the code prevents overflowing the buffer if the total length is greater than the buffer size).

If the current buffer size has been ok until now, then I think it is fine to keep its size as it is today. We don't need a larger buffer for the devices involved.